src/main/java/io/github/acm19/aws/interceptor/http/AwsRequestSigningApacheInterceptor.java

```diff
@@ -96,6 +94,11 @@ public void process(final HttpRequest request, final HttpContext context)
                 final ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
                 httpEntityEnclosingRequest.getEntity().writeTo(outputStream);
                 requestBuilder.contentStreamProvider(() -> new ByteArrayInputStream(outputStream.toByteArray()));
+                // copy back the entity, so it can be read again
+                BasicHttpEntity entity = new BasicHttpEntity();
+                entity.setContent(new ByteArrayInputStream(outputStream.toByteArray()));
+                // wrap into repeatable entity to support retries
+                httpEntityEnclosingRequest.setEntity(new BufferedHttpEntity(entity));
```

Comment:
Is there a more performant version that avoids a copy/read where we flip the order of these: 1) replace the entity with a repeatable entity, then set `requestBuilder.contentStreamProvider`?

Reply:
I changed it, please take a look now. I don't think it's substantially better, but hopefully it makes a bit more sense. I think it will be better if we change to exec interceptor as it's designed to change the entity. But I need to give it a try and see how it'd look like.

Reply:
Resolving and merging.